packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/ArbTokenBridge.sol

```diff
@@ -86,35 +93,99 @@ contract ArbTokenBridge is CloneFactory {
+
+        if(callHookData.length > 0) {
+            try ArbTokenBridge(this).mintAndCall(token, amount, sender, dest, callHookData) {
+                // the token's transfer hook does not get triggered here
+                // since the bridge already triggers a hook
+                emit MintAndCallTriggered(true);
+            } catch {
+                // if reverted, then credit sender's account
+                // TODO: should try to submit callHookData for 777's hook?
+                // this could become error prone since this is only run if dest reverts
+                token.bridgeMint(sender, amount, '');
+                emit MintAndCallTriggered(false);
+            }
+        } else {
+            // if no hook data, credit the dest
+            token.bridgeMint(dest, amount, '');
+        }
     }
 
     function mintERC20FromL1(
         address l1ERC20,
-        address account,
+        address sender,
+        address dest,
         uint256 amount,
-        uint8 decimals
+        uint8 decimals,
+        bytes calldata callHookData
     ) external onlyEthPair {
         IArbToken token = ensureERC20TokenExists(l1ERC20, decimals);
-        token.bridgeMint(account, amount);
+
+        if(callHookData.length > 0) {
+            try ArbTokenBridge(this).mintAndCall(token, amount, sender, dest, callHookData) {
+                emit MintAndCallTriggered(true);
+            } catch {
+                // if reverted, then credit sender's account
+                token.bridgeMint(sender, amount, '');
+                emit MintAndCallTriggered(false);
+            }
+        } else {
+            // if no hook data, credit the dest
+            token.bridgeMint(dest, amount, '');
+        }
     }
 
-    function mintCustomtokenFromL1(
+    function mintCustomTokenFromL1(
         address l1ERC20,
-        address account,
-        uint256 amount
+        address sender,
+        address dest,
+        uint256 amount,
+        bytes calldata callHookData
     ) external onlyEthPair {
         address tokenAddress = customToken[l1ERC20];
         require(tokenAddress != address(0), "Custom Token doesn't exist");
         IArbToken token = IArbToken(tokenAddress);
-        token.bridgeMint(account, amount);
+
+        if(callHookData.length > 0) {
+            try ArbTokenBridge(this).mintAndCall(token, amount, sender, dest, callHookData) {
+                // the token's transfer hook does not get triggered here
+                // since the bridge already triggers a hook
+                emit MintAndCallTriggered(true);
+            } catch {
+                // if reverted, then credit sender's account
+                // TODO: should try to submit callHookData?
+                token.bridgeMint(sender, amount, '');
+                emit MintAndCallTriggered(false);
+            }
+        } else {
+            // if no hook data, credit the dest
+            token.bridgeMint(dest, amount, '');
+        }
```

Comment:
Could these common blocks be refactored?

Reply:
We can